contracts/StableJoeStaking.sol

```diff
@@ -125,7 +135,11 @@ contract StableJoeStaking is Initializable, OwnableUpgradeable {
     function deposit(uint256 _amount) external {
         UserInfo storage user = userInfo[_msgSender()];
 
-        uint256 _fee = _amount.mul(depositFeePercent).div(DEPOSIT_FEE_PERCENT_PRECISION);
+        uint256 _fee;
+        // Only EOAs holding Smol Joes are exempt from paying the deposit fee
+        if (address(smolJoes) != 0 && (smolJoes.balanceOf(_msgSender()) == 0 || _msgSender() != tx.origin)) {
```

Comment:
Tests are failing as you are comparing address with int. 0 needs to be cast to address `address(0)`

Reply:
I think there is a potential for a problem here if you deploy this update and don't set the `smolJoe` variable immediately, as `&&` short circuits nobody would pay any fee on deposit until `smolJoe` address was set. It looks like this will be covered by the deployment script but still worth being aware.

Reply:
I think the following fixes the above problem.
```
if (!(address(smolJoes) != address(0) && smolJoes.balanceOf(_msgSender()) > 0 && _msgSender() == tx.origin)) {
     _fee = _amount.mul(depositFeePercent).div(DEPOSIT_FEE_PERCENT_PRECISION);
}
```